llama_index/question_gen/openai_generator.py

```diff
@@ -0,0 +1,84 @@
+from typing import List, Optional, Sequence, cast
+
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.llms.base import LLM
+from llama_index.llms.openai import OpenAI
+from llama_index.program.openai_program import OpenAIPydanticProgram
+from llama_index.question_gen.prompts import build_tools_text
+from llama_index.question_gen.types import (BaseQuestionGenerator, SubQuestion,
+                                            SubQuestionList)
+from llama_index.tools.types import ToolMetadata
+
+DEFAULT_MODEL_NAME = "gpt-3.5-turbo-0613"
+
+DEFAULT_OPENAI_SUB_QUESTION_PROMPT_TMPL = """\
+You are a world class state of the art agent.
+
+You have access to multiple tools, each representing a different data source or API.
+Each of the tools has a name and a description, formatted as JSON.
+Your purpose is to help answer a complex user question by generating a list of sub \
+questions that can be answered by the tools.
+
+These are the guidelines you consider when completing your task:
+* Be as specific as possible
+* The sub questions should be relevant to the user question 
+* The sub questions should be answerable by the tools provided
+* You can generate multiple sub questions for each tool
+* You don't need to use a tool if you don't think it's relevant
```

Comment:
Maybe out of scope here, but what happens if zero questions are generated?

Reply:
hmm, good question. not sure. may need to stress test this a bit more.

Reply:
if you force questions when there may not be it might hallucinate so you should either allow no questions to be asked or isntructinos on what the default behavior is.